feature/policy_forwarding/decapsulation/otg_tests/ipv4_guev1_decap_and_hashing_test/ipv4_guev1_decap_and_hashing_test.go

```diff
@@ -0,0 +1,1141 @@
+			flow := configureFlows(t, otgConfig, macAddress, []string{otgConfig.Ports().Items()[1].Name()}, flowIndex, false)
+			flows = append(flows, flow)
+		}
+		ate.OTG().PushConfig(t, otgConfig)
+		ate.OTG().StartProtocols(t)
+		ate.OTG().StartTraffic(t)
+		time.Sleep(trafficDuration * time.Second)
+		ate.OTG().StopTraffic(t)
+		for _, flow := range flows {
+			if ok := verifyFlowTraffic(t, ate, otgConfig, flow.Name()); !ok {
+				t.Fatalf("Packet loss detected in flow: %s", flow.Name())
+			} else {
+				t.Logf("Flow %s: Traffic validation sucess", flow.Name())
+			}
+		}
+	})
+	t.Run("PF-1.22.3: GUE Decapsulation over non-matching UDP decap port [Negative] test", func(t *testing.T) {
+		var flows []gosnappi.Flow
+		macAddress := gnmi.Get(t, dut, gnmi.OC().Interface(dut.Port(t, "port1").Name()).Ethernet().MacAddress().State())
+		otgConfig.Flows().Clear()
+		for flowIndex := 13; flowIndex <= 14; flowIndex++ {
+			flow := configureFlows(t, otgConfig, macAddress, []string{otgConfig.Ports().Items()[1].Name()}, flowIndex, false)
+			flows = append(flows, flow)
+		}
+		ate.OTG().PushConfig(t, otgConfig)
+		ate.OTG().StartProtocols(t)
+		ate.OTG().StartTraffic(t)
+		time.Sleep(trafficDuration * time.Second)
+		ate.OTG().StopTraffic(t)
+		for _, flow := range flows {
+			if ok := verifyTrafficFlowNegCase(t, ate, otgConfig, flow); !ok {
+				t.Logf("Flow %s: Packets dropped, Test Passed", flow.Name())
+			} else {
+				t.Fatalf("Flow %s: Packets not dropped, Test Failed", flow.Name())
+			}
+		}
+	})
+	t.Run("PF-1.22.4: Verify the Immediate next header's L4 fields are not considered in Load-Balancing Algorithm test", func(t *testing.T) {
+		t.Log("Starting test: Verify that immediate next header's L4 fields are NOT used in load-balancing")
+		macAddress := gnmi.Get(t, dut, gnmi.OC().Interface(dut.Port(t, "port1").Name()).Ethernet().MacAddress().State())
+		otgConfig.Flows().Clear()
+		// Generate flows with randomized L4 ports immediately after outer header
+		for flowIndex := 7; flowIndex <= 14; flowIndex++ {
+			configureFlows(t, otgConfig, macAddress, []string{otgConfig.Ports().Items()[1].Name()}, flowIndex, true)
+		}
+
+		ate.OTG().PushConfig(t, otgConfig)
+		ate.OTG().StartProtocols(t)
+		ate.OTG().StartTraffic(t)
+		time.Sleep(trafficDuration * time.Second)
+		ate.OTG().StopTraffic(t)
+
+		// Verify: Traffic should NOT be load-balanced → All traffic should go to a single port
+		verifySinglePathTraffic(t, ate, otgConfig)
+	})
+	t.Run("PF-1.22.5: Verify the Immediate next header's L3 fields are not considered in Load-Balancing Algorithm test", func(t *testing.T) {
+		t.Log("Starting test: Verify that immediate next header's L4 fields are NOT used in load-balancing")
+		macAddress := gnmi.Get(t, dut, gnmi.OC().Interface(dut.Port(t, "port1").Name()).Ethernet().MacAddress().State())
+		otgConfig.Flows().Clear()
+		// Generate flows with Immediate next header's L3 fields
+		for flowIndex := 1; flowIndex <= 14; flowIndex++ {
```

Comment:
```suggestion
		for flowIndex := 1; flowIndex <= 12; flowIndex++ {
```

I'll update the FNT as well

Reply:
ok, changes done